browser/brave_ads/creatives/search_result_ad/creative_search_result_ad_tab_helper.cc

```diff
@@ -166,12 +181,7 @@ void CreativeSearchResultAdTabHelper::MaybeHandleCreativeAdViewedEvents(
 void CreativeSearchResultAdTabHelper::MaybeHandleCreativeAdViewedEvent(
     const std::string& placement_id) {
   CHECK(!placement_id.empty());
-
-  // TODO(https://github.com/brave/brave-browser/issues/39404): Remove
-  // `base::Contains` and Sanitize JSON-LD placement id.
-  if (base::Contains(placement_id, '\"')) {
-    return;
-  }
+  CHECK(base::ranges::all_of(placement_id, &IsUnreservedCharOrPercentageSign));
```

Comment:
Check that placement id, which is passed to JS function contains only unreserved characters or percentage sign.